sklearn/metrics/pairwise.py

```diff
@@ -1139,18 +1140,27 @@ def cosine_distances(X, Y=None):
     array([[1.     , 1.     ],
            [0.42..., 0.18...]])
     """
+    xp, _ = get_namespace(X, Y)
+
     # 1.0 - cosine_similarity(X, Y) without copy
     S = cosine_similarity(X, Y)
     S *= -1
     S += 1
-    np.clip(S, 0, 2, out=S)
+    S = _clip(S, 0, 2, xp)
     if X is Y or Y is None:
         # Ensure that distances between vectors and themselves are set to 0.0.
         # This may not be the case due to floating point rounding errors.
-        np.fill_diagonal(S, 0.0)
+        _fill_diagonal_2d(S, 0.0, xp)
```

Comment:
In the _array_api utils we updated a function to support this `_fill_or_add_to_diagonal`
It can be used over here as follow
```suggestion
        _fill_or_add_to_diagonal(S, 0.0, xp, add_value=False)
```

Reply:
Done, thank you